### ISSUE TITLE
FileDialog filters don't work properly and nothing is loaded on first run 

### ISSUE THREAD

**Issue by ghost** [godotengine/godot#33533]
**Godot version:**
3.2 alpha 3

**OS/device including version:**
Windoze 10

**Issue description:**
FileDialog seems to be somewhat broken:
- filters get compounded in one long string if you keep switching between filters
- the pre-set folder/path is not preloaded on first run (nothing shows up until you click a filter could be specific, all or any)
- FilterDialog doesn't resize automatically when set to Full Rect. 

**Steps to reproduce:**

**Minimal reproduction project:**

First Run with preset root folder/path (nothing shows up)
![image](https://user-images.githubusercontent.com/36826319/68562896-9fce1c80-0419-11ea-8258-309d3655710f.png)

swapping between the two filters creates one long `*.*.*.*.whatever filter was clicked last`
![image](https://user-images.githubusercontent.com/36826319/68562927-c2603580-0419-11ea-8d8c-06b5e21fd20e.png)

**Comment by agameraaron** [comment]
All three of these bugs are confirmed as an issue for me as well in Linux [Mint 19.2] using the latest stable Godot Engine build [3.2.1].

**Comment by follower** [comment]
Seems (1) might be a dupe of https://github.com/godotengine/godot/issues/38195?

Which has an associated unmerged PR here: https://github.com/godotengine/godot/pull/38298

**Comment by rcorre** [comment]
> the pre-set folder/path is not preloaded on first run (nothing shows up until you click a filter could be specific, all or any)

Confirmed on Linux, with no filter. I'm using this as a workaround:

```
extends FileDialog

func _ready():
	connect("visibility_changed", self, "invalidate")
```

Just calling `invalidate` in `_ready` does not seem to work.

**Comment by marzecdawid** [comment]
Can this be closed as it is already fixed?

**Comment by akien-mga** [comment]
Fixed by #65389.
